dluhc-component-library/src/components/siteSelectionForm/SiteSelectionForm.tsx

```diff
@@ -28,25 +30,40 @@ interface SiteSelectionForm {
   uiSchema?: UiSchema;
 }
 
-const createValidationSchema = (key: string, formSchema: FormPageSchema) => {
-  let validationShape: ValidationShape | null = null;
-
-  const property = formSchema.properties?.[key];
+const REQUIRED_MESSAGE = "This field is required.";
 
+const getValidationShape: (property?: FormPageSchema) => ValidationShape = (
+  property,
+) => {
   switch (property?.type) {
-    case "string":
-      validationShape = string();
-      break;
     case "number":
-      validationShape = number();
-      break;
+      return number();
     case "boolean":
-      validationShape = boolean();
-      break;
+      return boolean();
+    case "array":
+      return array().of<any>(getValidationShape(property.items));
```

Comment:
This seems right to me, just double checking `any` is the type we want here?

Reply:
Yeah - we're having to use that to get around the Yup typing, not nice but Andy and I can't think of a better way